mm2src/mm2_main/src/lp_wallet.rs

```diff
@@ -0,0 +1,504 @@
+    #[display(fmt = "Wallets storage error: {}", _0)]
+    WalletsStorageError(String),
+    #[display(fmt = "Internal error: {}", _0)]
+    Internal(String),
+}
+
+impl HttpStatusCode for GetMnemonicError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            GetMnemonicError::InvalidRequest(_) => StatusCode::BAD_REQUEST,
+            GetMnemonicError::WalletsStorageError(_) | GetMnemonicError::Internal(_) => {
+                StatusCode::INTERNAL_SERVER_ERROR
+            },
+        }
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl From<WalletsStorageError> for GetMnemonicError {
+    fn from(e: WalletsStorageError) -> Self { GetMnemonicError::WalletsStorageError(e.to_string()) }
+}
+
+#[cfg(target_arch = "wasm32")]
+impl From<WalletsDBError> for GetMnemonicError {
+    fn from(e: WalletsDBError) -> Self { GetMnemonicError::WalletsStorageError(e.to_string()) }
+}
+
+impl From<ReadPassphraseError> for GetMnemonicError {
+    fn from(e: ReadPassphraseError) -> Self { GetMnemonicError::WalletsStorageError(e.to_string()) }
+}
+
+/// Retrieves the wallet mnemonic in the requested format.
+///
+/// # Returns
+///
+/// A `Result` type containing:
+///
+/// * [`Ok`]([`GetMnemonicResponse`]) - The wallet mnemonic in the requested format.
+/// * [`MmError`]<[`GetMnemonicError>`]> - Returns specific [`GetMnemonicError`] variants for different failure scenarios.
+///
+/// # Errors
+///
+/// This function will return an error in the following situations:
+///
+/// * The wallet name is not found in the context.
+/// * The wallet is initialized without a name.
+/// * The wallet passphrase file is not found for `MnemonicFormat::Encrypted`.
+/// * The wallet mnemonic file is not found for `MnemonicFormat::PlainText`.
+///
+/// # Examples
+///
+/// ```rust
+/// let ctx = MmArc::new(MmCtx::default());
+/// let req = GetMnemonicRequest {
+///     mnemonic_format: MnemonicFormat::Encrypted,
+/// };
+/// let result = get_mnemonic_rpc(ctx, req).await;
+/// match result {
+///     Ok(response) => println!("Mnemonic: {:?}", response.mnemonic),
+///     Err(e) => println!("Error: {:?}", e),
+/// }
+/// ```
+pub async fn get_mnemonic_rpc(ctx: MmArc, req: GetMnemonicRequest) -> MmResult<GetMnemonicResponse, GetMnemonicError> {
+    match req.mnemonic_format {
+        MnemonicFormat::Encrypted => {
+            let encrypted_mnemonic = read_encrypted_passphrase_if_available(&ctx)
+                .await?
+                .ok_or_else(|| GetMnemonicError::InvalidRequest("Wallet passphrase file not found".to_string()))?;
+            Ok(GetMnemonicResponse {
+                mnemonic: encrypted_mnemonic.into(),
+            })
+        },
+        MnemonicFormat::PlainText(wallet_password) => {
+            let plaintext_mnemonic = read_and_decrypt_passphrase_if_available(&ctx, &wallet_password)
+                .await?
+                .ok_or_else(|| GetMnemonicError::InvalidRequest("Wallet mnemonic file not found".to_string()))?;
```

Comment:
I guess both 'Wallet passphrase file' and 'Wallet mnemonic file' are actually the same file. I think this error may confuse the user and maybe it is better to return the same error description in both cases?

Reply:
Done